include/ONV.hpp

```diff
@@ -0,0 +1,144 @@
+#ifndef GQCG_ONV_HPP
+#define GQCG_ONV_HPP
+
+
+#include <Eigen/Dense>
+#include "common.hpp"
+
+
+
+namespace GQCG {
+
+
+class ONV {
+private:
+    const size_t K;  // number of spatial orbitals
+    const size_t N;  // number of electrons
+    size_t unsigned_representation;  // unsigned representation
+    VectorXs occupation_indexes;  // the occupied orbital electron indexes
+
+
+public:
+    // CONSTRUCTORS
+    /**
+     *  Constructor from a @param K orbitals, N electrons and a representation for the ONV
+     */
+    ONV(size_t K, size_t N, size_t unsigned_representation);
+
+
+    // OPERATORS
+    /**
+     *  Overloading of operator<< for a GQCG::ONV to be used with streams
+     */
+    friend std::ostream& operator<<(std::ostream& os, const GQCG::ONV& onv);
+
+    /**
+     *  @return if this equals @param other
+     */
+    bool operator==(ONV& other) const {
+        return this->unsigned_representation == other.unsigned_representation && this->K == other.K;  // this ensures that N, K and representation are equal
+    }
+
+    /**
+     *  @return if this is not equal to @param other
+     */
+    bool operator!=(ONV& other) const {
+        return !(this->operator==(other));
+    }
+
+
+    // GETTERS & SETTERS
+    void set_representation(size_t unsigned_representation);
+    size_t get_urepresentation(){ return unsigned_representation;}
+    VectorXs get_occupations(){ return occupation_indexes;}
+
+    /**
+     *  @return occupied orbital based on the electron index
+     */
+    size_t get_occupied_orbital(size_t electron_index);
+
+
+    // PUBLIC METHODS
+    /**
+     *  Extracts the positions of the set bits from the representation and places them in an array
+     */
+    void update();
+
+
+    // PUBLIC METHODS (PREVIOUS SPIN STRING FUNCTIONALITY)
+    /**
+     *  @return if the index is occupied (i.e. 1) for the @param p-th spatial orbital, starting from 0
+     *  @param p is the lexical index (i.e. read from right to left)
+     */
+    bool isOccupied(size_t p) const;
+
+    /**
+     *  @return if we can apply the annihilation operator (i.e. 1->0) for the @param p-th spatial orbital, starting from 0
+     *  performs an annihilation @param p
+     *
+     *  !!! IMPORTANT: does not update the occupation array if required call "update()" !!!
+     *  !!! IMPORTANT: performs the annihilation in place !!!
+     */
+    bool annihilate(size_t p);
+
+    /**
+     *  @return if we can apply the annihilation operator (i.e. 1->0) for the @param p-th spatial orbital, starting from 0
+     *  @param p is the lexical index (i.e. read from right to left)
+     *
+     *  Furthermore, the @param sign is changed according to the sign change (+1 or -1) of the spin string after annihilation.
+     *
+     *  !!! IMPORTANT: does not update the occupation array if required call "update()" !!!
+     *  !!! IMPORTANT: performs the annihilation in place !!!
+     */
+    bool annihilate(size_t p, int& sign);
+
+    /**
```

Comment:
For the creation operators, see the comments on the annihilation operators to make the documentation consistent